src/NBitcoin/Block.cs

```diff
@@ -357,7 +368,7 @@ public bool Check()
         /// <returns></returns>
         public bool Check(Consensus consensus)
         {
-            if (Block.BlockSignature)
+            if ((consensus?.NetworkOptions ?? NetworkOptions.TemporaryOptions).IsProofOfStake)
```

Comment:
Same comment  as previously

Reply:
Here we are recognizing the fact that consensus itself may be null as per the legacy behavior of this method. The NetworkOptions already defaults to TemporaryOptions on the Consensus object.